mkosi/__init__.py

```diff
@@ -4994,15 +4991,13 @@ def run_verb(args: Args, images: Sequence[Config], *, resources: Path) -> None:
 
     last = images[-1]
 
-    if not in_sandbox() and last.tools_tree and last.tools_tree == Path("default"):
+    if last.tools_tree and last.tools_tree == Path("default"):
         tools = finalize_default_tools(last, resources=resources)
     else:
         tools = None
 
     for i, config in enumerate(images):
-        if in_sandbox():
-            tools_tree = None
-        elif tools and config.tools_tree == Path("default"):
+        if tools and config.tools_tree == Path("default"):
             tools_tree = tools.output_dir_or_cwd() / tools.output
         else:
             tools_tree = config.tools_tree
```

Comment:
mypy is complaining about this, but since it figures `tools_tree` should be `Path`. Adding `tools_tree: Optional[Path]` is one way to solve this, since it was obviously nullable before.